docs/docs/learn/shards/coding-with-shards.md

```diff
@@ -95,138 +213,421 @@ In summary:
+
+![A Wire is made up of a sequence of shards.](assets/what-is-a-wire.png)
+
+To create a Wire, we use [`defwire`](../../../reference/functions/macros/#defwire).
+
+=== "Creating a Wire"
     
     ```{.clojure .annotate linenums="1"}
-    (Repeat 
-        (-> (Msg "Hello World!")) ;; (1)
-        :Times 2)
+    (defwire wire-name 
+      ;; shards here
+    )
     ```
 
-    1. You can still implicitly declare the first parameter, while fully declaring the other parameters. Note that it does not work vice versa. You cannot implicitly declare parameters if the parameter before it has been fully declared.
+!!! note
+    The syntax for `defwire` is different from `defshards` as you cannot define parameters. Square brackets `[]` are not used.
 
-=== "3 Parameters (Fully Declared)"
+!!! note
+    Unlike `defshards` which groups shards up for organization, `defwire` groups shards up to fulfill a purpose. As Wires are created with a purpose in mind, they should be appropriately named to reflect it.
+
+A Wire's lifetime ends once the final shard within it has been executed. To keep a Wire alive even after it has reached its end, we can set it to be loopable. This is called a Looped Wire.
+
+![A Looped Wire is kept alive even after the final shard is executed.](assets/what-is-a-looped-wire.png)
+
+A Looped Wire will continue running until exit conditions have been met.
+
+!!! note
+    You will learn more about the entering and exiting of Looped Wires in the next chapter!
+
+To create a Looped Wire, we use [`defloop`](../../../reference/functions/macros/#defloop).
+
+=== "Creating a Looped Wire"
     
     ```{.clojure .annotate linenums="1"}
-      0 >= .x ;; (6)
-  
+    (defloop loop-name 
+      ;; shards here
+    )
+    ```
+
+## The Mesh
+
+Wires are queued for execution within a Mesh, from left to right, top to bottom.
+
+![Wires are queued for execution within a Mesh.](assets/what-is-a-mesh.png)
+
+To queue a Wire on a Mesh, we use [`schedule`](../../../reference/functions/misc/#schedule).
+
+=== "Scheduling a Wire"
+    
+    ```{.clojure .annotate linenums="1"}
+    (schedule mesh-name wire-name)
+    ```
+
+!!! note
+    We will learn more about controlling the flow of Shards with Wires and Meshes in the following chapter.
+
+
+## Running Shards
+
+In order to actually get Shards running, a specific hierarchy and sequence must be followed. Your shards are first queued into Wires, which are then queued onto a Mesh.
+
+![The hierarchy of a Shards program.](assets/shards-hierarchy.png)
+
+When the Mesh is run, the Wires are executed in sequence and your program is started. This is done using the aptly named command [`run`](../../../reference/functions/misc/#run).
+
+=== "Running a Mesh"
+    
+    ```{.clojure .annotate linenums="1"}
+    (run mesh-name)
+    ```
+
+`run` can take in two optional values:
+
+- The interval between each iteration of the Mesh.
+
+- The maximum number of iterations, used only for Debugging purposes.
```

Comment:
It might be useful in other case, so I would rephrase it.

```suggestion
- The maximum number of iterations, typically used for Debugging purposes.
```